applications/vanilla/models/class.categorymodel.php

```diff
@@ -285,6 +295,96 @@ public static function getRootDisplayAs() {
         return c('Vanilla.RootCategory.DisplayAs', 'Categories');
     }
 
+    /**
+     * Get a list of a user's followed categories.
+     *
+     * @param int $userID The target user's ID.
+     * @return array
+     */
+    public function getFollowed($userID) {
+        $key = "Follow_{$userID}";
+        $result = Gdn::cache()->get($key);
+        if ($result === Gdn_Cache::CACHEOP_FAILURE) {
+            $sql = clone $this->SQL;
+            $sql->reset();
+
+            $userData = $sql->getWhere('UserCategory', [
+                'UserID' => $userID,
+                'Follow' => 1
+            ])->resultArray();
+            $result = array_column($userData, null, 'CategoryID');
+            Gdn::cache()->store($key, $result);
+        }
+
+        return $result;
+    }
+
+    /**
+     * Set whether a user is following a category.
+     *
+     * @param int $userID The target user's ID.
+     * @param int $categoryID The target category's ID.
+     * @param bool|null $follow True for following. False for not following. Null for toggle.
+     * @return bool A boolean value representing the user's resulting "follow" status for the category.
+     */
+    public function follow($userID, $categoryID, $follow = null) {
+        $validationOptions = ['options' => [
+            'min_range' => 1
+        ]];
+        if (!($userID = filter_var($userID, FILTER_VALIDATE_INT, $validationOptions))) {
+            throw new InvalidArgumentException('Invalid $userID');
+        }
+        if (!($categoryID = filter_var($categoryID, FILTER_VALIDATE_INT, $validationOptions))) {
+            throw new InvalidArgumentException('Invalid $categoryID');
+        }
+
+        $category = static::categories($categoryID);
+        if (!is_array($category)) {
+            throw new InvalidArgumentException('Category not found.');
+        } elseif ($category['DisplayAs'] !== 'Discussions') {
```

Comment:
Unfollowing should work here. (unit test that)

Reply:
Makes good sense. Done.